link-grammar/print/print.c

```diff
@@ -743,8 +769,8 @@ char * linkage_print_diagram(const Linkage linkage, bool display_walls, size_t s
 	ps_ctxt_t ctx;
 	if (!linkage) return NULL;
 
-	ctx.link_heights = (int *) alloca(linkage->num_links * sizeof(int));
-	ctx.row_starts = (int *) alloca((linkage->num_words + 1) * sizeof(int));
+	ctx.link_heights = NULL;
```

Comment:
??? Cough cough. How can this possibly be correct?

Reply:
`link_heights` and `row_starts` are need only for the postscript generation. So for the ASCII diagram they can just be ignored.

Reply:
ah I see now.